tests/savepoint/conftest.py

```diff
@@ -401,8 +423,25 @@ def get_parallel_param(
 
 
 def get_parallel_mock_param(
-    case, testobj, savepoint_in_list, savepoint_out_list, call_count, max_call_count
+    case,
+    testobj,
+    savepoint_in_list,
+    savepoint_out_list,
+    call_count,
+    max_call_count,
+    only_one_rank,
 ):
+    dependency = (
+        pytest.mark.dependency()
+        if only_one_rank
+        else pytest.mark.dependency(
```

Comment:
this might produce strange results with the mock parallel option. Most of these do halo updates (just in loops), so running only one rank will definitely give the wrong answer. But it's not a huge deal to leave it as is.

Reply:
I am going to raise a `RuntimeError` on mock_parallel and parallel if `--which_rank` is used